pkg/informer/informer.go

```diff
@@ -108,10 +109,47 @@ func (d *DynamicDiscoverySharedInformerFactory) informerForResourceLockHeld(gvr
 		nil,
 	)
 
+	inf.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: d.filterFunc,
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				d.handlerLock.RLock()
+
+				for _, h := range d.handlers {
+					h.OnAdd(gvr, obj)
+				}
+
+				d.handlerLock.RUnlock()
```

Comment:
this is not good. Calling out to foreign code under a lock is a nogo. Why not make `d.handlers` update copy on write?

Reply:
Thanks for the suggestion - will fix